### PR TITLE
Make address selection searchable

### LINKED ISSUE #2141: Address selection (transaction sending), should be searchable
- by name
- by address
- (all in addition to the lookups & copy/paste)
- not 100% certain what the UI does, but needs to be simple and uncluterred

(Split from https://github.com/ethcore/parity/issues/2114)

### PR DESCRIPTION
**Pull request by ngotchac**
Fixes #2141 

Implements address selection in transaction modal with an autocomplete, so they can be searched through...

See it in action : https://youtu.be/C-xgoemIs7c

### CODE REVIEW

File: js/src/ui/Form/InputAddress/inputAddress.js [hunk 1]
```diff
@@ -75,6 +93,13 @@ class InputAddress extends Component {
       </div>
     );
   }
+
+  handleInputChange = (event, value) => {
+    let isEmpty = (value.length === 0);
```
Comment (jacogr) [inline js/src/ui/Form/InputAddress/inputAddress.js]: const? ;)

File: js/src/ui/Form/AddressSelect/addressSelect.js [hunk 2]
```diff
@@ -36,31 +35,75 @@ export default class AddressSelect extends Component {
...
-        disabled={ disabled }
-        label={ label }
-        hint={ hint }
-        error={ error }
-        value={ value }
-        onChange={ this.onChange }>
-        { this.renderSelectEntries() }
-      </Select>
+      <div>
+        <AutoComplete
```
Comment (jacogr) [inline js/src/ui/Form/AddressSelect/addressSelect.js]: Could we (for the sake of consistency in the input components), add the AutoComplete to where we have the other input components, i.e. a thin wrapper. It feels like the odd man out atm.

**Comment by jacogr** [comment]
Small comments, minor issues. Fix and we can push it in.

File: js/src/ui/Form/InputAddress/inputAddress.js (hunk 1 above)
Comment (ngotchac) [inline js/src/ui/Form/InputAddress/inputAddress.js]: Right, still some sticking old habits...

File: js/src/ui/Form/AddressSelect/addressSelect.js [hunk 3]
```diff
@@ -89,7 +132,45 @@ export default class AddressSelect extends Component {
...
+    }
+
+    const entries = Object.values(this.state.entries);
+    const address = entries[idx].address;
+
+    this.props.onChange(null, address);
+  }
+
+  onFocus = () => {
+    this.props.onChange(null, '');
```
Comment (ngotchac) [inline js/src/ui/Form/AddressSelect/addressSelect.js]: Yep, agreed. `event` should be only within Inputs wrappers

File: js/src/ui/Form/InputAddress/inputAddress.js (hunk 1 above)
Comment (jacogr) [inline js/src/ui/Form/InputAddress/inputAddress.js]: :D It happens.

**Pull request merged**